### PR TITLE
feat: Added Split expenses restrictions in add edit expense

### PR DESCRIPTION
**Pull request by howdysuraj**
### Description
Added Split expenses restrictions in add edit expense

## Clickup
https://app.clickup.com/t/86cup8d7v

## UI Preview
https://www.loom.com/share/c2e5e4e3668d453d9a85e31dbe13dc23?sid=d70686d9-be6c-4074-bf58-fd17e11a4373

### CODE REVIEW

File: src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts [hunk 1]
```diff
@@ -1116,6 +1116,8 @@ export function TestCases1(getTestBed) {
 
         spyOn(component, 'openSplitExpenseModal');
 
+        component.pendingTransactionAllowedToReportAndSplit = true;
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts:1119]: can we set this in `beforeEach` in the describe block and then use `component.pendingTransactionAllowedToReportAndSplit = true;` wherever required
You can set this to true in beforeEach in line 1114

File: src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts [hunk 2]
```diff
@@ -1124,16 +1126,30 @@ export function TestCases1(getTestBed) {
...
+
+        expect(component.showTransactionPendingToast).toHaveBeenCalledTimes(1);
+      });
     });
 
     describe('splitExpProjectHandler():', () => {
       it('should call method to display split expense modal and split by project', () => {
         setFormValid();
 
+        component.pendingTransactionAllowedToReportAndSplit = true;
```
Comment (suyashpatil78) [inline src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts:1151]: same as above

**Review by suyashpatil78 — APPROVED** [review]
minor comments! rest LGTM 🍏

File: src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts (hunk 1 above)
Comment (howdysuraj) [reply src/app/fyle/add-edit-expense/add-edit-expense-1.spec.ts]: Done

**Pull request merged**